pkg/handler/collector/deps_dev/deps_dev.go

```diff
@@ -84,31 +86,36 @@ type depsCollector struct {
 
 var registerOnce sync.Once
 
-func NewDepsCollector(ctx context.Context, collectDataSource datasource.CollectSource, poll, retrieveDependencies bool, interval time.Duration, addedLatency *time.Duration) (*depsCollector, error) {
+func NewDepsCollector(ctx context.Context, collectDataSource datasource.CollectSource, poll, retrieveDependencies bool, interval time.Duration, addedLatency *time.Duration, rateLimitedClient grpc.ClientConnInterface) (*depsCollector, error) {
```

Comment:
Don't add the option to pass in a client if we are not using it anywhere in the code.

Reply:
We pass in the rate limiter for when we test the code.

https://github.com/guacsec/guac/blob/1aca21c629cf1e2ad5642c366b1245e6742abfd9/pkg/handler/collector/deps_dev/deps_dev.go#L98-L99

Reply:
I am not sure I understand what you mean

Reply:
Ohh... I see what you mean. I have fixed the issue.